parser/src/ast/callable.rs

```diff
@@ -5,8 +5,12 @@ use lexer::token::Token;
 /// A call to a function or a command.
 #[derive(Debug, Clone, PartialEq)]
 pub struct Call<'a> {
-    /// command name at index 0 and its arguments
+    /// The arguments of the command.
+    ///
+    /// A valid command must have at least one argument that is the command name.
     pub arguments: Vec<Expr<'a>>,
+    /// The redirections of the command.
+    pub redirections: Vec<Redir<'a>>,
```

Comment:
I Think you should move the redirections of a call out of the `Call` structure because it'll be much easier in the future to handle redirections as an `Expr -> Redir` binding instead of a collection of different kinds of redirection for the call. 
Moreover, this current data structure only allows calls to have redirection, but RFC#2 says that this is possible for any kind of expressions, (a block, a match, a loop etc)